src/components/EventsTimeline/components/ChartLayout.tsx

```diff
@@ -0,0 +1,226 @@
+import { CogniteEvent } from '@cognite/sdk';
+import { event as d3Event } from 'd3';
+import { select } from 'd3-selection';
+import {
+  zoom as d3Zoom,
+  ZoomBehavior,
+  ZoomedElementBaseType,
+  ZoomScale,
+} from 'd3-zoom';
+import { debounce, Dictionary } from 'lodash';
+import React, { RefObject, useEffect, useState } from 'react';
+import { getEventsByTimestamp, getScaleTime } from '../helpers';
+import { EventTimelineType, EventTimelineView } from './Event';
+import { Ruler } from './Ruler';
+import { Timeline } from './Timeline';
+
+interface TimelineEventAppearance {
+  view: EventTimelineView;
+  type: EventTimelineType;
+}
+
+export interface TimelineEvent extends TimelineEventAppearance {
+  id: number;
+}
+
+export interface TimelineRuler {
+  show: boolean;
+  onChange?: (event: React.SyntheticEvent, date: number) => void;
+  onEventHover?: (event: CogniteEventForTimeline[] | null) => void;
+  hoverDebounceTime?: number;
```

Comment:
I think we don't need to expose this one, and will do it if it's requested by users later